eventsdiscovery/src/main/java/com/schedjoules/client/eventsdiscovery/http/CategoriesArrayResponseHandler.java

```diff
@@ -39,20 +40,28 @@
  *
  * @author Gabor Keszthelyi
  */
-public final class CategoriesArrayResponseHandler implements HttpResponseHandler<Iterator<Category>>
+public final class CategoriesArrayResponseHandler implements HttpResponseHandler<Iterable<Category>>
 {
     @Override
-    public Iterator<Category> handleResponse(final HttpResponse response) throws IOException, ProtocolError, ProtocolException
+    public Iterable<Category> handleResponse(final HttpResponse response) throws IOException, ProtocolError, ProtocolException
     {
-        JSONArray categoriesArray = new JSONArray(new StringResponseHandler().handleResponse(response));
-        return new Mapped<>(new JsonArrayIterator(categoriesArray),
-                new Function<JSONObject, Category>()
-                {
-                    @Override
-                    public Category apply(JSONObject element)
-                    {
-                        return new JsonCategory(element);
-                    }
-                });
+        final JSONArray categoriesArray = new JSONArray(new StringResponseHandler().handleResponse(response));
+
+        return new Iterable<Category>()
+        {
+            @Override
+            public Iterator<Category> iterator()
+            {
+                return new Mapped<>(new JsonArrayIterator(categoriesArray),
```

Comment:
No need to create an anonymous `Iterable`. Instead of `JsonArrayIterator` use `JsonArrayIterable` and the new `Mapped` `Iterable` decorator.